content/knowledge_base/consensus/avalanche/specification.md

```diff
@@ -0,0 +1,44 @@
+# Avalanche
+
+An internet-scale electronic payment system, evaluated in a large scale deployment, constructed with Snow family of algorithms. Avalance's tested confirmation latency goes down to 1.35 sec. System's bottleneck is transaction verification.
+TODO: elaborate more on the introduction
+
+## Characteristics
+
+Avalanche network operates with multiple single-decree instances (nodes), meaning that each instance can make a decision on only one value. Nodes use Snowball algorithm for the process of making a decision. (TODO: add a reference) In the case of a payment system and Avalanche, the Snowball logic used to determine a value is a process of deciding on a transaction being valid or not. 
+
+The protocol maintains the set of all known transactions using a dynamic, append-only Directed Acyclic Graph (DAG) structure. (TODO: add reference)
+There are two main positives from this:
+ * DAG helps streamlining the path on which there are no conflicting transactions - a single vote on a DAG vertex implicitly votes for all transactions on the path to the genesis vertex
+ * using the same principle as BTC, the DAG tree renders past decisions on the path of transactions as more and more difficult to undo without approval of the correct nodes
+Genesis block concept is implemented with the DAG having a single sink - "genesis vertex".
+
+When a new transaction is created its parents are defined in the DAG structure as its ancestors, forming the edges with their newly created decendant. The parent-child relationship in DAG is not mandatory for a child transaction may not have any relation to its parents transactions outputs. It can actually spend funds received in another transactions from its "ancestor set". The "ancestor set" is the set of all transactions reachable via parent edges back in history (TODO: add reference to the quote). Oppositely - the term "progeny" means all existing (and potentially existing) children transactions and their children.
+
+Main purpose of a consensus protocol is to avoid inclusion of conflicting transactions into the ledger. Even though a conflicting transaction characteristics are defined on an application level, the notion of conflict can be abstracted in order to define the conflicting set. In Avalanche every transaction belongs to a conflict set, meaning every transaction is a part of a set of transactions that contains one (if the set is a singleton set - the conflicting transactions will be potentially created or the singleton set contains a single virtuos transaction) or more transactions that are invalid towards each other existence.
```

Comment:
```suggestion
The main purpose of a consensus protocol is to avoid the inclusion of conflicting transactions into the ledger.
Even though a conflicting transaction's characteristics are defined on an application level, the notion of conflict can be abstracted in order to define the conflicting set.
In Avalanche, every transaction belongs to a conflict set, meaning every transaction is a part of a set of transactions that contains one (if the set is a singleton set, the conflicting transactions will be potentially created or the singleton set contains a single virtual transaction) or more transactions that are invalid towards each other existence.
```

I guess you meant virtual transaction?

Reply:
No, it was just unclear, updated this part, IMO is clear now, WDYT?
https://github.com/optakt/optakt.github.io/pull/13/commits/0a1cb902f3417c3bc20f9197df71d3d65a1b6421#diff-69f45991c0732c6f5ff297a5a86ffa91662d50369fc8057b3875361f5b80a2fcR32-R34

Reply:
It's much better yep :) Thanks!